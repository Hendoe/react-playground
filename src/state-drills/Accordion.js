import React from 'react';




  //this.sections.[2]
  //set hidden based on 


//classname dependednt on if clicked = true or false 
class Accordion extends React.Component{

    static defaultProps = { sections : []};

    state = {
        activeSectionIndex: null

    };

    handleButtonClick(index){
        this.setState({activeSectionIndex : index})
    }

    render() {

        const buttons = this.props.sections.map((section, index) => (
                <li>
                <button key={index} onClick={this.handleButtonClick}>
                    {section.title}
                </button>
                </li> 
        ));
        const currentSection = this.props.sections[0];
        return (
                <ul>
                   {buttons}
                   { }
                   {this.props.sections.length && (
                   <div className ='content'>
                       {currentSection.content}
                   </div>
                   )}
                </ul>
            
        )
    }


    

//prop named sections
//object containing title and content each





}













export default {
    Accordion
};