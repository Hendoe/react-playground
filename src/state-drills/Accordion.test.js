import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';

import renderer from 'react-test-renderer';

describe('Accordion component', () => {


it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
});


it('renders UI as expected', () => {
    const tree = renderer
    .create(<Accordion title="title" content="content" />)
    .toJSON();
expect(tree).toMatchSnapshot();
})







});