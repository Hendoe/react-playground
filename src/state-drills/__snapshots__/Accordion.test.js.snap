// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component renders UI as expected 1`] = `
<ul>
  0
</ul>
`;
